src/chrome/content/ff-overlay.js

```diff
@@ -85,30 +85,37 @@ if (typeof SL_Overlay === "undefined") {
                 .addEventListener("popupshowing", (event) => {
                     this.showFirefoxContextMenu(event);
                 }, false);
-            DownloadManager.connectToNas();
         };
 
         this.onMenuItemLinkCommand = (event) => {
-            if (DownloadManager.isConnected) {
-                window.open(DownloadManager.urlToConnect + "/webman/index.cgi?launchApp=SYNO.SDS.DownloadStation.Application", "Diskstation", this.strWindowFeatures);
-            } else {
-                Notification.show("No Connection");
-            }
+            DownloadManager.connectAndRun(() => {
+                if (DownloadManager.isConnected) {
+                    window.open(DownloadManager.urlToConnect + "/webman/index.cgi?launchApp=SYNO.SDS.DownloadStation.Application", "Diskstation", this.strWindowFeatures);
+                } else {
+                    Notification.show("No Connection");
+                }
+            });
         };
 
         this.onMenuItemCommand = (event) => {
-            if (DownloadManager.isConnected) {
-                DownloadManager.transferToNas(gContextMenu.linkURL);
-            } else {
-                Notification.show("No Connection");
-            }
+            let linkURL = gContextMenu.linkURL;
+            DownloadManager.connectAndRun(() => {
```

Comment:
I understand what you mean.

The idea behind it is this: Connect to the DiskStation, then run the command we're busy with.
As it's used in different places, I was looking for something generic.

What would you suggest?